proxy/healthy_endpoints_test.go

```diff
@@ -241,3 +244,39 @@ func TestPHC(t *testing.T) {
 		})
 	}
 }
+
+func TestPHCMaxUnhealthyEndpointsEdgeCase(t *testing.T) {
+	const (
+		healthy   = 2
+		unhealthy = 2
+	)
+
+	servicesString := setupServices(t, healthy, unhealthy)
+	endpointregistry := routing.NewEndpointRegistry(routing.RegistryOptions{
+		PassiveHealthCheckEnabled:     true,
+		StatsResetPeriod:              period,
+		MinRequests:                   2,
+		MaxHealthCheckDropProbability: 0.95,
+		MinHealthCheckDropProbability: 0.01,
+		MaxUnhealthyEndpointsRatio:    0.5, // because of this setting PHC expected effectively not to work here
```

Comment:
I think the edge case is when this is 1.0, then
```go
unhealthyEndpointsCount == len(endpoints)*e.Metrics.MaxUnhealthyEndpointsRatio()
```

Reply:
Yeah, I just realized that the "edge case" is "all endpoints are unhealthy", implemented such case instead here:
https://github.com/zalando/skipper/pull/3081/files#diff-b1bbc8a9c24c3b87d0de99e08ebcdd6df2da583d27fa530049a3cc12d496435cR248